scripts/test-remote-dev-codex-runtime.py

```
@@ -272,6 +308,213 @@ def test_successful_publication_retains_previous_generation(self):
+            with mock.patch.object(self.m, "STAGING_ROOT", fixed), mock.patch.object(
+                self.m, "probe_staging_execution"
+            ):
+                with self.m.update_staging() as staging:
+                    package_bin = staging / "package" / "bin"
+                    package_bin.mkdir(parents=True)
+                    (staging / "package").chmod(0o755)
+                    candidate = package_bin / "codex"
+                    candidate.write_text("#!/bin/sh\nexit 0\n", encoding="utf-8")
+                    candidate.chmod(0o755)
+                    package_bin.chmod(0o600)
+                    with self.assertRaisesRegex(
```

Ruff (0.16.1)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)